### ISSUE TITLE
as a researcher, I'd like to resolve persistent identifiers in a way that isn't vulnerable to MITM

### ISSUE THREAD

**Issue by pameyer** [IQSS/dataverse#4469]
"copy link" on the "doi:$PID" or "hdl:$PID" (in the citation block) gives a `http` url; looks like it should give `https`

DataCite recommendations for DOIs (https://support.datacite.org/v1.1/docs/datacite-doi-display-guidelines) recommend that the resolver URL be `https://doi.org/$DOI` (instead of `http://dx.doi.org/$DOI`).

Unsure of recommendations for handles, but https URLS there appear to resolve correctly.

Related (although less extensive than) #4237

**Comment by pdurbin** [comment]
> recommend that the resolver URL be https://doi.org/$DOI (instead of http://dx.doi.org/$DOI).

Thanks, I was wondering about this when looking at pull request #4470 (which I just approved) before I had looked at this issue. I guess I should read the issue first. 😄

**Comment by pdurbin** [comment]
@matthew-a-dunlap and I were just talking about if all of the instances of `dx.doi` should be changed. This is from develop (as of 6dd1fcb), not the pull request:

```
(dataverse) murphy:dataverse pdurbin$ ack dx.doi src
src/main/java/edu/harvard/iq/dataverse/GlobalId.java
26:    public static final String DOI_RESOLVER_URL = "http://dx.doi.org/";

src/main/java/edu/harvard/iq/dataverse/GlobalId.java.sekmiller
81:               url = new URL("http://dx.doi.org/" + authority + "/" + identifier); 

src/main/java/edu/harvard/iq/dataverse/search/SearchFields.java
86:     * Such as http://dx.doi.org/10.5072/FK2/HXI35W

src/test/java/edu/harvard/iq/dataverse/DatasetVersionTest.java
124:        assertEquals("http://dx.doi.org/10.5072/FK2/LK0D1H", obj.getString("identifier"));

src/test/java/edu/harvard/iq/dataverse/export/ddi/codebook.xsd
3884:<IDNo>http://dx.doi.org/10.3886/ICPSR32547.v1</IDNo>

src/test/java/edu/harvard/iq/dataverse/export/ddi/dataset-finch-private.xml
13:      <biblCit>Finch, Fiona, 2015, "Darwin's Finches", http://dx.doi.org/10.5072/FK2/PCA2E3,  Root Dataverse,  V1</biblCit>

src/test/java/edu/harvard/iq/dataverse/export/ddi/dataset-finch1.json
4:  "persistentUrl": "http://dx.doi.org/10.5072/FK2/PCA2E3",
380:    "citation": "Finch, Fiona, 2015, \"Darwin's Finches\", http://dx.doi.org/10.5072/FK2/PCA2E3,  Root Dataverse,  V1"

src/test/java/edu/harvard/iq/dataverse/export/ddi/dataset-finch1.xml
13:      <biblCit>Finch, Fiona, 2015, "Darwin's Finches", http://dx.doi.org/10.5072/FK2/PCA2E3,  Root Dataverse,  V1</biblCit>

src/test/java/edu/harvard/iq/dataverse/export/ddi/dataset-finchDC.xml
4:  <dcterms:identifier>http://dx.doi.org/10.5072/FK2/PCA2E3</dcterms:identifier>

src/test/java/edu/harvard/iq/dataverse/export/ddi/dataset-spruce1.json
4:  "persistentUrl": "http://dx.doi.org/10.5072/FK2/OZDVMO",
105:    "citation": "Spruce, Sabrina, 2015, \"Spruce Goose\", http://dx.doi.org/10.5072/FK2/OZDVMO,  Root Dataverse,  V1

src/test/java/edu/harvard/iq/dataverse/export/ddi/dataset-spruce1.xml
13:      <biblCit>Spruce, Sabrina, 2015, "Spruce Goose", http://dx.doi.org/10.5072/FK2/OZDVMO,  Root Dataverse,  V1</biblCit>

src/test/java/edu/harvard/iq/dataverse/export/ddi/datasetContactEmailAbsent.json
7:    "citation": "Finch, Fiona, 2016, \"Darwin's Finches\", http://dx.doi.org/10.5072/FK2/G2VPE7, Root Dataverse, V1",

src/test/java/edu/harvard/iq/dataverse/export/ddi/datasetContactEmailPresent.json
7:    "citation": "Finch, Fiona, 2016, \"Darwin's Finches\", http://dx.doi.org/10.5072/FK2/G2VPE7, Root Dataverse, V1",

src/test/java/edu/harvard/iq/dataverse/export/SchemaDotOrgExporterTest.java
162:        assertEquals("http://dx.doi.org/myAuthority/myIdentifier", json2.getString("identifier"));

src/test/java/edu/harvard/iq/dataverse/util/json/jsondata.txt
272:        "persistentUrl": "http://dx.doi.org/10.5072/FK2/19",

src/test/resources/json/complete-dataset-with-files-invalid-directory-labels.json
119:                  "value": "http://dx.doi.org/10.1016/j.cell.2011.10.020"

src/test/resources/json/complete-dataset-with-files.json
119:                    "value": "http://dx.doi.org/10.1016/j.cell.2011.10.020"
(dataverse) murphy:dataverse pdurbin$ 
```

`src/main/java/edu/harvard/iq/dataverse/GlobalId.java` is taken care of already in the pull request.

I assume `src/main/java/edu/harvard/iq/dataverse/GlobalId.java.sekmiller` can safely be deleted.

The rest are comments and test files. It's probably not necessary to change them all, in my opinion.

**Comment by scolapasta** [comment]
Discussed with @kcondon and @djbrooke and we're moving back to dev to clean up the remaining dx.doi.org references (from test files and comments).

**Comment by kcondon** [comment]
I did test functionality:
citations on file, dataset, and dataset cards for hdl and doi. works but need to reindex all to fix urls in cards

**Comment by pdurbin** [comment]
> moving back to dev to clean up the remaining dx.doi.org references (from test files and comments)

Fixed in d7dc91e and docker-aio test suite passes (using instructions from e796935). Moving to Code Review at https://waffle.io/IQSS/dataverse
